### PR TITLE
[prometheus-node-exporter] custom servicemonitor apiversion

### PR DESCRIPTION
**Pull request by lusson-luo**
Signed-off-by: lusson <lusson@foxmail.com>

#### What this PR does / why we need it
Add values prometheus.monitor.apiVersion to modify serviceMonitor api.
To meet the needs of modifying the apiversion of serviceMonitor to access the monitoring platforms of some cloud manufacturers.

#### Which issue this PR fixes

no

- fixes #

#### Checklist

- [x] [DCO](https://github.com/prometheus-community/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [x] Chart Version bumped
- [x] Title of the PR starts with chart name (e.g. `[prometheus-couchdb-exporter]`)

### CODE REVIEW

**Review by zanhsieh — DISMISSED** [review]

**Comment by zanhsieh** [comment]
@lusson-luo 
Could you sign DCO please? Thank you.

File: charts/prometheus-node-exporter/values.yaml [hunk 1]
```diff
@@ -28,7 +28,7 @@ env: {}
 
 prometheus:
   monitor:
-    enabled: false
+    enabled: true
```
Comment (monotek) [inline charts/prometheus-node-exporter/values.yaml:31]: Unrelated change?
Would break update. 
Please remove.

**Review by monotek — CHANGES_REQUESTED** [review]

**Comment by lusson-luo** [comment]
> @lusson-luo Could you sign DCO please? Thank you.

ok，pass the DCO

**Review by zanhsieh — CHANGES_REQUESTED** [review]
@lusson-luo 
Could you fix this lint error please?

**Review by lusson-luo — COMMENTED** [review]
i have review my change

File: charts/prometheus-node-exporter/templates/servicemonitor.yaml [hunk 2]
```diff
@@ -1,5 +1,5 @@
 {{- if .Values.prometheus.monitor.enabled }}
-apiVersion: monitoring.coreos.com/v1
+apiVersion: {{ .Values.prometheus.monitor.apiVersion | default "monitoring.coreos.com/v1" }}
```
Comment (monotek) [inline charts/prometheus-node-exporter/templates/servicemonitor.yaml:2]: Imho this could be more generic. Instead if defining API version manually it could be checked which API version is available and the most recent one is used.

**Review by monotek — CHANGES_REQUESTED** [review]

File: charts/prometheus-node-exporter/templates/servicemonitor.yaml (hunk 2 above)
Comment (zanhsieh) [reply charts/prometheus-node-exporter/templates/servicemonitor.yaml:2]: I feel @lusson-luo intends to forcibly override apiVersion for matrix build purposes. Auto detect which apiVersion available is less concerned.

Comment (lusson-luo) [reply charts/prometheus-node-exporter/templates/servicemonitor.yaml:2]: > what api exactly is wanted? if its just version change it could be done automatically.
> 
> Something like:
> 
> `{{- if .Capabilities.APIVersions.Has "route.openshift.io/v1/Route" }}`

@monotek The scenario is like this, 
I encapsulate a node-exporter helm chart, which will be deployed on the k8s bases of various cloud manufacturers. The monitoring platforms of cloud manufacturers are different, but most of them will provide servicemonitors with promethues-like rules, but apiversion does not The same. When the operation and maintenance personnel deploy my charts, they can adapt only by modifying the values, without changing the helm source code.

**Review by zanhsieh — APPROVED** [review]

**Review by monotek — APPROVED** [review]
